library/ecp.c

```diff
@@ -2061,6 +2061,60 @@ int mbedtls_ecp_check_pub_priv( const mbedtls_ecp_keypair *pub, const mbedtls_ec
     return( ret );
 }
 
+#if defined(MBEDTLS_ASN1_WRITE_C) && defined(MBEDTLS_OID_C)
+#include "mbedtls/asn1write.h"
```

Comment:
Please move those `#include` directive near the top of the file with the others (and the appropriate `#if` obviously). Burying such directives in the middle of the file is a departure from our coding style, and IMO it's useful to be able to see all of them at a glance by looking at the top of the file.

Reply:
Done.